### DISCUSSION TITLE
New user confirmation/activation mail

### DISCUSSION THREAD

**Discussion by wm-simon** [statamic/cms#4441] [General]
I think it's a common technic to send mails with an activation link for new user registrations. I couldn't find something in the docs to handle this. Is there a way to add this feature? Should this not be the default behavior?

**Comment by jasonvarga** [comment]
It's in there. Create a user through the control panel and it gives you the option to send an email.

<img width="788" alt="image" src="https://user-images.githubusercontent.com/105211/136800676-043c40f4-e5bc-465a-97b7-c8aa4ae21958.png">

**Comment by wm-simon** [comment]
The users create their account via the frontend, perhaps that's an important info ;-)
`{{ user:register_form }}`

Or is it possible to send this mail with the register_form tag?

**Reply by jasonvarga** [reply]
Why do they need an email then? They entered their password. They're activated.

**Reply by wm-simon** [reply]
To be sure the user email is correct. If the not activated users would be deleted after a specific time automatically, there would be less spam-users in the system. I know this procedure from other other systems. I think on this way it's handled in nearly every shop-system, where new users are often created. Perhaps I'm wrong, it's just my experience.

**Reply by jasonvarga** [reply]
There is a `Statamic\Events\UserRegistered` event that is dispatched after registration. You can create a listener for that that'll send an email.

**Reply by wm-simon** [reply]
I'm not a back dev, but I try my best with the listener.
```
<?php

namespace App\Listeners;

use Illuminate\Contracts\Queue\ShouldQueue;
use Illuminate\Queue\InteractsWithQueue;
use Statamic\Events\UserRegistered;

use App\User;

class registrationMail
{
    /**
     * Create the event listener.
     *
     * @return void
     */
    public function __construct()
    {
        //
    }

    /**
     * Handle the event.
     *
     * @param  object  $event
     * @return void
     */
    public function handle(UserRegistered $event)
    {
            $user = $event->user;
            Mail::send('emails.welcome', ['user' => $user], function ($message) use ($user) {
            $message->from('status@domain.de', 'Website');
            $message->subject('Registrierung erfolgreich '.$user->first_name.' '.$user->last_name.'!');
            $message->to($user->email); 
        });
    }

}

```
There happens nothing. Is this all wrong?

**Reply by duncanmcclean** [reply]
Did you register your listener in your `EventServiceProvider`?
